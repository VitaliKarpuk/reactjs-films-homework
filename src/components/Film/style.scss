.movie-list__item {
  width: 500px;
  height: 760px;
  background-color: rgb(255, 255, 255);
  margin: 10px;
  border-radius: 15px;
  -webkit-border-radius: 15px;
  -moz-border-radius: 15px;
  -ms-border-radius: 15px;
  -o-border-radius: 15px;
  position: relative;

  img {
    width: 100%;
    height: 600px;
    border-top-left-radius: 15px;
    border-top-right-radius: 15px;
  }

  .movie-list__item_img {
    width: 100%;
    height: 600px;
    font-size: 28px;
    letter-spacing: 1px;
    line-height: 44px;
    color: #242b2d;
    font-weight: bold;
    background-color: rgb(201, 201, 201);
    border-top-left-radius: 15px;
    border-top-right-radius: 15px;
  }

  .movie-list__rating {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 30px 20px 0 20px;
    overflow: hidden;

    h3 {
      font-size: 28px;
      letter-spacing: 1px;
      line-height: 44px;
      color: #242b2d;
      font-weight: bold;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    span {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 60px;
      height: 36px;
      border-radius: 6px;
      border: 2px solid #49c1f0;
      font-size: 24px;
      line-height: 44px;
      color: #47c3f0;
      font-weight: 500;
      text-align: center;
    }
  }

  .movie-list__genre{
    font-size: 22px;
    letter-spacing: 1px;
    line-height: 33px;
    color: #47c3f0;
    font-weight: 500;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    padding: 20px;
  }

}
