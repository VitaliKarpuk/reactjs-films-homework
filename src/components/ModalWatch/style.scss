.movie-hover {
  width: 100%;
  height: 600px;
  position: absolute;
  top: 0;
  border-top-left-radius: 15px;
  border-top-right-radius: 15px;
  background-color: rgba(0, 0, 0, 0.781);
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  opacity: 0;
  transition: all 0.3s;
  -webkit-transition: all 0.3s;
  -moz-transition: all 0.3s;
  -ms-transition: all 0.3s;
  -o-transition: all 0.3s;

  h3 {
    font-size: 28px;
    letter-spacing: 1px;
    line-height: 44px;
    color: #ffffff;
    text-align: center;
    margin: 45px 0;
  }

  .movie-hover__icon{
    width: 117px;
    height: 117px;
    position: relative;
    border-radius: 50%;
    -webkit-border-radius: 50%;
    -moz-border-radius: 50%;
    -ms-border-radius: 50%;
    -o-border-radius: 50%;
    background-color: #3bb4e0;
    transition: all 0.3s;
    -webkit-transition: all 0.3s;
    -moz-transition: all 0.3s;
    -ms-transition: all 0.3s;
    -o-transition: all 0.3s;
    margin-top: 180px;

    &::after {
      content: "";
      position: absolute;
      left: 48px;
      top: 28px;
      border: 30px solid transparent;
      border-left: 35px solid rgb(255, 255, 255);
    }

    &:hover {
      cursor: pointer;
      background-color: #1d7ea1;
    }
  }

  &:hover {
    opacity: 1;
  }
}
