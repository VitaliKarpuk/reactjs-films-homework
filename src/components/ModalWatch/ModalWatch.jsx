import React from "react";
import PropTypes from "prop-types";

import "./style.scss";

const ModalWatch = ({ handleBtnInfo, handleShowrailer, movieListView }) => {
  return (
    <div className="movie-hover">
      <div className="movie-hover__icon" onClick={handleShowrailer}></div>
      <h3>Watch Now</h3>
      {
        movieListView && <button className="btn__info" onClick={handleBtnInfo}>View Info</button>
      }
      
    </div>
  );
};

ModalWatch.propTypes = {
  handleShowrailer: PropTypes.func,
  handleBtnInfo: PropTypes.func,
  movieListView: PropTypes.bool
};

ModalWatch.defaultProps = {
  handleShowrailer: () => {},
  handleBtnInfo: () => {},
  movieListView: true
};

export default ModalWatch;
