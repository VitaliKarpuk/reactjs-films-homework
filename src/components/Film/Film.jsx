import React, { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import PropTypes from "prop-types";
import ModalWatch from "../ModalWatch/ModalWatch";
import ModalInfoFilm from "../ModalInfoFilm/ModalInfoFilm";
import ModalTrailer from "../ModalTrailer/ModalTrailer";
import requesTrailer from "../../modules/actions/requesTrailer";
import { selectorTrailer } from "../../modules/selectors/index";

import "./style.scss";

const Film = ({
  title,
  overview,
  imgUrl,
  rating,
  genreFilmArray,
  changeGenre,
  genre,
  id,
  movieListView
}) => {
  const [showInfoFilm, setShowInfoFilm] = useState(false);
  const [genreFilm, setGenreFilm] = useState([]);
  const [showTrailer, setShowTrailer] = useState(false);
  const trailer = useSelector(state => selectorTrailer(state));
  const dispatch = useDispatch();

  useEffect(() => {
    if (genreFilmArray && genre.length > 0) {
      setGenreFilm(changeGenre(genreFilmArray));
    }
  }, [genreFilmArray, genre]);

  const toggleInfoModal = () => setShowInfoFilm(showInfoFilm => !showInfoFilm);

  const toggleTrailerModal = () => {
    !showTrailer && dispatch(requesTrailer(id));
    setShowTrailer(showTrailer => !showTrailer);
  };

  return (
    <div className="movie-list__item">
      {imgUrl ? (
        <img
          src={`https://image.tmdb.org/t/p/w600_and_h900_bestv2/${imgUrl}`}
          alt={`${title}`}
        />
      ) : (
        <div className="movie-list__item_img">{title}</div>
      )}
      <div className="wrapper__rating">
        <div className="movie-list__rating">
          <h3>{title}</h3>
          <span>{rating}</span>
        </div>
        <p className="movie-list__genre">
          {genreFilm.map((item, index) =>
            genreFilm[index + 1] ? `${item}, ` : `${item}`
          )}
        </p>
        {!movieListView && <p className="movie-list__overview">{overview}</p>}
      </div>

      {showInfoFilm ? (
        <ModalInfoFilm
          title={title}
          rating={rating}
          genreFilm={genreFilm}
          overview={overview}
          handleCloseInfo={toggleInfoModal}
          handleShowrailer={toggleTrailerModal}
        />
      ) : (
        <ModalWatch
          handleBtnInfo={toggleInfoModal}
          handleShowrailer={toggleTrailerModal}
          showTrailer={showTrailer}
          movieListView={movieListView}
        />
      )}
      {showTrailer && (
        <ModalTrailer
          className={"btn__info"}
          title={title}
          showTrailer={showTrailer}
          handleCloseTrailer={toggleTrailerModal}
          trailer={trailer}
        />
      )}
    </div>
  );
};

Film.propTypes = {
  title: PropTypes.string,
  overview: PropTypes.string,
  imgUrl: PropTypes.string,
  rating: PropTypes.number,
  id: PropTypes.number,
  genreFilmArray: PropTypes.array.isRequired,
  genre: PropTypes.array,
  changeGenre: PropTypes.func,
  movieListView: PropTypes.bool
};

Film.defaultProps = {
  title: "",
  overview: "",
  imgUrl: "",
  rating: 0,
  id: Math.random(),
  genre: [],
  changeGenre: () => {},
  movieListView: true
};

export default Film;
