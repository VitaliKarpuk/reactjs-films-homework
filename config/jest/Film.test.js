/* eslint-disable no-undef */
import React from 'react'
import { render, unmountComponentAtNode } from "react-dom";
import TestRenderer, { act, create } from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import ReactTestUtils from 'react-dom/test-utils';
import Film from '../../src/components/Film/Film';
import ModalWatch from '../../src/components/ModalWatch/ModalWatch';

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('Test Film component', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Film />);
  const result = renderer.getRenderOutput();
  expect(result.type).toBe('div');
  expect(result.props.className).toBe('movie-list__item');
  expect(result.props.children[0].type).toBe('div');
  expect(result.props.children[1].type).toBe('div');
  expect(result.props.children[2].type).toBe('p');
  expect(result.props.children[4]).toMatchObject(<ModalWatch />);
  expect(result.props.children.length).toBe(5)

  expect(result).toMatchSnapshot();
})

const tree = create(<Film />)
it('Test Film component', () => {
  tree.toJSON();
  expect(tree).toMatchSnapshot();
  console.log(tree)
  expect(tree).toMatchSnapshot();
  // act(() => {
  //   render(<Film />, container);
  // });
  // const div = container.querySelector('.movie-list__item');
  // console.log(div);

  // ReactTestUtils.Simulate.change(input, { target: { value: 'test' } });
  // expect(input.value).toBe("test");
  // ReactTestUtils.Simulate.keyDown(input, { key: "E" });

})

it('test', () => {
  const button = tree.root.findAllByType('button');
  // act( ()=> button.onPress())
  console.log(button.getIs);
})