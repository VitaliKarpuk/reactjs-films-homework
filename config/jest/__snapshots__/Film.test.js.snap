// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component renders Films component 1`] = `
<div
  className="movie-list__item"
>
  <div
    className="movie-list__item_img"
  >
    
  </div>
  <div
    className="wrapper__rating"
  >
    <div
      className="movie-list__rating"
    >
      <h3>
        
      </h3>
      <span>
        0
      </span>
    </div>
    <p
      className="movie-list__genre"
    />
  </div>
  <div
    className="movie-hover"
  >
    <div
      className="movie-hover__icon"
      onClick={[Function]}
    />
    <h3>
      Watch Now
    </h3>
    <button
      className="btn__info"
      onClick={[Function]}
    >
      View Info
    </button>
  </div>
</div>
`;
